src/sd.jl

```diff
@@ -96,6 +97,40 @@ function vectorize(matrix::Matrix, ::SymmetricMatrixShape)
     return [matrix[i, j] for j in 1:n for i in 1:j]
 end
 
+
+"""
+    SkewSymmetricMatrixShape
+
+Shape object for a skew symmetric square matrix of `side_dimension` rows and columns.
+The vectorized form contains the entries of the upper-right triangular part of
+the matrix (without the diagonal) given column by column (or equivalently, the
+entries of the lower-left triangular part given row by row). The diagonal is zero.
+"""
+struct SkewSymmetricMatrixShape <: AbstractShape
+    side_dimension::Int
+end
+function reshape_vector(vectorized_form::Vector{T}, shape::SkewSymmetricMatrixShape) where T
+    matrix = Matrix{JuMP.GenericAffExpr{Float64,JuMP.VariableRef}}(undef, shape.side_dimension, shape.side_dimension)
```

Comment:
It should be `Matrix{Base.promote_type(T, MA.promote_operation(-, T), MA.promote_operation(zero, T))}` so that it works for any `T`

Reply:
thanks!